File: JAVA/1669. Merge In Between Linked Lists.java
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode list2_lastNode=list2;
        while(list2_lastNode.next!=null) list2_lastNode=list2_lastNode.next;

        ListNode criticalNode=list1;
        int temp_a=a;
        while(true){
            temp_a--;
            if(temp_a<=0) break;
            criticalNode=criticalNode.next;
        }
        b=b-a+1;
        ListNode tempcriticalNode=criticalNode;
        while(tempcriticalNode.next!=null){
            if(b==0) break;
            tempcriticalNode=tempcriticalNode.next;
            b--;
        }
        criticalNode.next=list2;
        list2_lastNode.next=tempcriticalNode.next;

        return list1;
    }
}

File: C++/1669. Merge In Between Linked Lists.cpp
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode *start,*last,*head=list1;
        int num=1;
        while(list1 != NULL){
            if(num==a){
                start=list1;
            }
            else if(num==b+2){
                last=list1;
            }
            list1=list1->next;
            num++;
        }
        start->next=list2;
        while(start->next!=NULL){
            start=start->next;
        }
        start->next=last;
        return head;
    }
};
